src/main/java/com/boot/user/service/UserService.java

```diff
@@ -179,13 +184,19 @@ public GetUserDTO getUserByEmail(String email, boolean includeDetails, boolean i
 
     @Transactional
     public void deleteUserByEmail(String email) {
-        log.info("deleteUserByEmail - process started");
+        User user = userRepository.getUserByEmail(email);
 
-        if (userRepository.getUserByEmail(email) == null) {
+        if (user == null) {
             throw new EntityNotFoundException("Email: " + email + " not found in the Database!");
         }
 
-        userRepository.deleteByEmail(email);
+        ResponseEntity response = cartServiceClient.deleteCartByUserId(user.getId());
+        log.info("deleteCartByUserId - process started");
+
+        if (response.getStatusCode().name().equals(HttpStatus.OK.name())) {
```

Comment:
response.getStatusCode().is2xxSuccessful() would be more elegant

Reply:
nice suggestion 🌟